lib/rack/method_override.rb

```diff
@@ -46,7 +46,7 @@ def allowed_methods
     end
 
     def method_override_param(req)
-      req.POST[METHOD_OVERRIDE_PARAM_KEY]
+      req.POST[METHOD_OVERRIDE_PARAM_KEY] if req.form_data? || req.parseable_data?
```

Comment:
I will look myself, but if you already know you can help me, there is no significant performance cost of calling these methods right? (i.e. it's not parsing the form body etc).

Reply:
The methods just check the request method and headers, they don't deal with the body.